src/kudu/consensus/raft_consensus.h

```diff
@@ -333,6 +333,11 @@ class RaftConsensus : public std::enable_shared_from_this<RaftConsensus>,
   // Returns the current term.
   int64_t CurrentTerm() const;
 
+  // Returns uuid of the current leader
+  const std::string& CurrentLeaderUUID() const {
```

Comment:
Maybe you need to just make `GetLeaderUuidUnlocked` public?
Taking the mutex might be required for this read since a write might be in flight.

Reply:
Makes sense, I couldn't find GetLeaderUuidUnlocked :)

Reply:
we should not return by reference, but by value.